src/data_structure/ziplist/ziplist.c

```diff
@@ -353,12 +389,11 @@ ziplist_remove_val(uint32_t *removed, ziplist_p zl, const struct blob *val,
         return ZIPLIST_EINVALID;
     }
 
-    if (count == 0) {
-        return ZIPLIST_EINVALID;
-    }
-
+    count = count == 0 ? INT64_MAX : count;
     atmost = forward ? count : -count;
-    *removed = 0;
+    if (removed != NULL) {
```

Comment:
Good catch guarding against empty pointer. Shall we use an internal variable and assign that to `*removed` at the end, to avoid extra jump instructions?